internal/commands/root/http.go

```diff
@@ -98,6 +98,15 @@ func setupHTTPServer(ctx context.Context, p provider.Provider, cfg *apiServerCon
 		}
 		api.AttachPodRoutes(podRoutes, mux, true)
 
+		var summaryHandlerFunc api.PodStatsSummaryHandlerFunc
+		if mp, ok := p.(provider.PodMetricsProvider); ok {
+			summaryHandlerFunc = mp.GetStatsSummary
+		}
+		podMetricsRoutes := api.PodMetricsConfig{
+			GetStatsSummary: summaryHandlerFunc,
+		}
+		api.AttachPodMetricsRoutes(podMetricsRoutes, mux)
```

Comment:
What happens if summaryHandlerFunc is null because the provider doesn't implement pod metrics provider?

Reply:
We return a `NotImplemented` error:

https://github.com/virtual-kubelet/virtual-kubelet/blob/b50302b8451937dd5b9badfe46559529f1b10f8d/node/api/stats.go#L31-L33